modyn/tests/conftest.py

```diff
@@ -0,0 +1,186 @@
+# pylint: disable=redefined-outer-name
```

Comment:
Previously we handed over technically invalid configuration dictionaries all the time. Fields that were marked as required in the jsonschema weren't added in the tests configuration just assuming that the implementation wouldn't need them. Now with pydantic we enforce the schema all the time.

Reply:
Well, let's say I often have been lazy when writing the tests... Thank you for cleaning this up.

Reply:
> Well, let's say I often have been lazy when writing the tests... Thank you for cleaning this up.

I mean for a research environment that's a fair approach at the beginning to evaluate things fast. It makes it harder though to adjust previous design decisions. And it can cause more debugging efforts when you don't have fields available that your typing suggests you have.